snorkel/labeling/lf.py

```diff
@@ -6,6 +6,53 @@
 
 
 class LabelingFunction:
+    """Base object for labeling functions.
+
+    A labeling function (LF) is a function that takes a data point
+    as input and produces an integer label, corresponding to a
+    class. A labeling function can also abstain from voting by
+    outputting 0. For examples, see the Snorkel tutorials.
+
+    This object wraps a Python function outputting a label. Metadata
+    about the input data types and label space are stored. Extra
+    functionality, such as running preprocessors and storing
+    resources, is provided. Simple LFs can be defined via a
+    decorator. See `labeling_function`.
+
+    Parameters
+    ----------
+    name
+        Name of the LF
+    f
+        Function that implements the core LF logic
+    label_space
+        Set of labels the LF can output, including 0. By default None.
```

Comment:
We've got a lot of "by default None" statements in the docstrings right now, and that's already specified by the explicit typing. I'd vote to drop all these?